use-timescale/compression/compression-design.md

```diff
@@ -0,0 +1,176 @@
+looks like this instead:
+
+|Timestamp|Device ID|Status Code|Temperature|
+|-|-|-|-|
+|12:00:01|A|0|70.11|
+|0 seconds|B|0|69.70|
+|1 second|A|0|70.12|
+|0 seconds|B|0|69.69|
+|1 second|A|0|70.14|
+|0 seconds|B|4|69.70|
+
+This compresses the timestamps to 64 bits plus 2 * 20 bits, which is 73%
+compression, and much higher than the first example, for exactly the same data.
+
+By default, TimescaleDB orders data by time, but you can also add additional
+columns for ordering.
+
+## Segmenting compressed data
+
+When you have compressed your data in the way described in the previous section,
+the database has no way of knowing which rows need to be fetched and
+decompressed to resolve a query. For example, the database can't easily
+determine which rows contain data from the past day, as the timestamp itself is
+in a compressed column. You don't want to have to decompress all the data in a
+chunk, or even an entire hypertable, to determine which rows are required.
+TimescaleDB automatically includes more information in each batch to improve
+query performance.
+
+For order-by columns, TimescaleDB automatically adds additional data that stores
+the minimum and maximum value of that column. This way, the query planner can
+look at the range of timestamps in the compressed column, without having to
+decompress a batch, and determine whether the data in the batch could possibly
+match the query.
+
+However, using the previous example, if you wanted to query the data based on
+device ID, you would have to decompress the device column to decide if that
+batch should be part of the query. When you compress your hypertable, you can
+choose to specify a `SEGMENT BY` column. This allows you to segment compressed
+rows by a specific column, so that each compressed row corresponds to a data
+about a single item such as, for example, a specific device ID. For example, you
+could use `SEGMENT BY` to create batches like this:
+
+|Device ID|Timestamp|Status Code|Temperature|Min Timestamp|Max Timestamp|
+|-|-|-|-|-|-|
+|A|[12:00:01, 12:00:02, 12:00:03]|[0, 0, 0]|[70.11, 70.12, 70.14]|12:00:01|12:00:03|
+|B|[12:00:01, 12:00:02, 12:00:03]|[0, 0, 0]|[70.11, 70.12, 70.14]|12:00:01|12:00:03|
+
+With the data segmented this way, a query for device A in a particular time
+interval becomes quite fast. The query planner can use an index to find those
+rows for device A that contain at least some timestamps corresponding to the
+specified interval, and even a sequential scan is quite fast, since evaluating
+device IDs or timestamps does not require decompression. This means the query
+executor only decompresses the timestamp and temperature columns corresponding
+to those selected rows.
+
+This, however, comes at the expense of a worse compression ratio. In the
+example, you now have two rows with three items in each compressed column rather
+than a single row with six items in the compressed rows.
```

Comment:
This is only true if the batches contain less than 1000 values, so if each device has more than 1000 values inside a chunk, it should still compress well. You might want to elaborate on this a little more.